### PR TITLE
Add telemetry.* attributes to resource

### LINKED ISSUE #1272: Add telemetry.* resource attributes
# Feature Request

Enable the `telemetry.*` resource attributes that are defined in the [resource semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk).

**Is your feature request related to a problem? Please describe.**
Enabling these resource attributes will help us to differentiate which SDK and auto-instrumentation version is being used to gather the telemetry. This will also help us to differentiate SDK usage from auto-instrumentation usage.

**Describe the solution you'd like**
We should be able to enable the out-of-the box `telemetry.sdk.*` attributes by using code already provided by the SDK. We should also add the `telemetry.auto.version` attribute to match the version of auto-instrumentation being used.

**Describe alternatives you've considered**
None

**Additional context**
The OTel java agent is already providing these attributes, and we should provide these attributes too.

### PR DESCRIPTION
**Pull request by nrcventura**
## Why

Adds helpful attributes that are defined in the semantic conventions for resources.

Fixes #1272

## What

Adds the telemetry.sdk attributes using the helper method in the SDK, and manually adds the telemetry.auto.version attribute using the auto-instrumentation version info.

## Tests

Adds a new smoke test specifically for validating resource attributes.

## Checklist

- [x] `CHANGELOG.md` is updated.
- ~~[ ] Documentation is updated.~~
- [x] New features are covered by tests.

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM. 
There is another PR which proposed changes `MockMetricsCollector`. We need to ensure `ResourceTests` has necessary changes based on whichever gets merged first.

**Review by pjanotti — APPROVED** [review]
👍

File: src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs [hunk 1]
```diff
@@ -110,6 +111,10 @@ public static void Initialize()
                 }
 
                 EnvironmentConfigurationSdkHelper.UseEnvironmentVariables(SdkSettings);
+
+                _resourceBuilder
+                    .AddTelemetrySdk()
+                    .AddAttributes(new[] { new KeyValuePair<string, object>(Constants.Tracer.AutoInstrumentationVersionName, Constants.Tracer.Version) });
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:117]: supreme nit 😬
```suggestion
                    .AddAttributes(new KeyValuePair<string, object>[] { new(Constants.Tracer.AutoInstrumentationVersionName, Constants.Tracer.Version) });
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Support `OTEL_DOTNET_AUTO_GRAPHQL_SET_DOCUMENT` (default value: `false`)
   environment variable which controls whether `graphql.document` attribute
   is set.
+- Add [telemetry resource attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/77856ebe90ea7503c793a3140a1dce3383f45126/specification/resource/semantic_conventions#telemetry-sdk).
```
Comment (Kielek) [inline CHANGELOG.md:22]:
```suggestion
- Add [telemetry resource attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.13.0/specification/resource/semantic_conventions#telemetry-sdk).
```

**Review by Kielek — APPROVED** [review]

**Comment by pellared** [comment]
> LGTM. There is another PR which proposed changes `MockMetricsCollector`. We need to ensure `ResourceTests` has necessary changes based on whichever gets merged first.

This should help https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1280

**Comment by pellared** [comment]
@nrcventura I updated the PR. I hope you are fine with it 😉

**Review by Kielek — APPROVED** [review]

File: test/IntegrationTests/SmokeTests.cs [hunk 3]
```diff
@@ -133,6 +134,10 @@ public async Task ResourceMetrics()
     {
         using var collector = await MockMetricsCollector.Start(Output);
         collector.ExpectResourceAttribute("service.name", ServiceName);
+        collector.ExpectResourceAttribute("telemetry.sdk.name", "opentelemetry");
+        collector.ExpectResourceAttribute("telemetry.sdk.language", "dotnet");
+        collector.ExpectResourceAttribute("telemetry.sdk.version", typeof(OpenTelemetry.Resources.Resource).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version);
```
[the remark this answers is not in the document]
Comment (nrcventura) [reply test/IntegrationTests/SmokeTests.cs:139]: I almost did not include it too. The main reason I kept it is that we have been talking about loading 2 different SDK versions. So we will likely want a test to validate that the correct version is loaded, but we might want that to be a different test.

**Comment by nrcventura** [comment]
> @nrcventura I updated the PR. I hope you are fine with it 😉

Thank you for updating and merging the PR. You did exactly what I would have done 👍.

**Pull request merged**